m4a-dec.c

```diff
@@ -121,21 +118,16 @@ int main(int argc, char *argv[]) {
 			}
 			frame_size = info->frameSize * info->numChannels;
 			// Note, this probably doesn't return channels > 2 in the right order for wav
-			wav = wav_write_open(outfile, info->sampleRate, 16, info->numChannels);
+			wav = wav_write_open(outfile, info->sampleRate, sizeof(INT_PCM)*8, info->numChannels);
 			if (!wav) {
 				perror(outfile);
 				status = 1;
 				break;
 			}
 		}
-		for (i = 0; i < frame_size; i++) {
-			uint8_t* out = &output_buf[2*i];
-			out[0] = decode_buf[i] & 0xff;
-			out[1] = decode_buf[i] >> 8;
-		}
```

Comment:
The old code here intentionally converted from native endian (which the decoder uses) to little endian when writing to a file. Previously this example wrote proper wav files even when run on a big endian system, now it no longer does.

Reply:
Ok, not a problem to return endian back if you find it is important. I'll fix.